wavefront_sdk/common/metrics/registry.py

```diff
@@ -41,7 +41,7 @@ def _schedule_timer(self):
     # pylint: disable=broad-except
     def _report(self, timeout_secs=None):
         timestamp = time.time()
-        for key, val in self.metrics.items():
+        for key, val in self.metrics.copy().items():
```

Comment:
Consider adding a comment explaining why this change is necessary. Just explain that copying the dict here protects from concurrent modification while iterating.

Reply:
Done.